dymos/utils/introspection.py

```diff
@@ -774,119 +779,118 @@ def configure_timeseries_output_introspection(phase):
-            expr_kwargs[var]['units'] = var_units
-        elif var == expr_lhs:
-            meta['units'] = expr_kwargs[var]['units']
-
-        if 'shape' in expr_kwargs[var] and var != expr_lhs:
-            om.issue_warning(f'{phase.pathname}: User-provided shape for timeseries expression input `{var}` ignored.\n'
-                             f'Using automatically determined shape {(num_output_nodes,) + var_shape}.',
-                             category=om.UnusedOptionWarning)
-        elif 'shape' in expr_kwargs[var] and var == expr_lhs:
-            if expr_options['shape'] is not _unspecified:
-                om.issue_warning(f'{phase.pathname}: User-provided shape for timeseries expression output `{var}`'
-                                 f'using both the\n`shape` keyword argument and the `{var}` keyword argument '
-                                 f'dictionary.\nThe latter will take precedence.',
-                                 category=om.SetupWarning)
-            var_shape = expr_kwargs[var]['shape']
-            meta['shape'] = var_shape
-
-        expr_kwargs[var]['shape'] = var_shape
-        kwargs_rel_shape[var] = {'units': expr_kwargs[var]['units'],
-                                 'shape': (num_output_nodes,) + expr_kwargs[var]['shape']}
-
-    for input_var in expr_vars:
-        abs_name = phase.timeseries_ec_vars[input_var]['abs_name']
-        if '.' in abs_name:
-            expr_reduced = expr_reduced.replace(abs_name, input_var)
-
-    timeseries_ec.add_expr(expr_reduced, **kwargs_rel_shape)
-
-    for var, options in expr_vars.items():
-        if phase.timeseries_ec_vars[var]['added_to_ec']:
-            continue
-        else:
-            phase.connect(src_name=options['src'], tgt_name=f'{timeseries_name}.timeseries_exec_comp.{var}',
-                          src_indices=options['src_idxs'])
-
-        phase.timeseries_ec_vars[var]['added_to_ec'] = True
+    for ts_name, ts_opts in phase._timeseries.items():
+        timeseries_ec = phase._get_subsystem(f'{ts_name}.timeseries_exec_comp')
+        phase.timeseries_ec_vars[ts_name] = {}
 
-    meta['src_idxs'] = None
+        for output_name, output_options in ts_opts['outputs'].items():
+            if output_options['is_expr']:
 
-    return meta
+                expr = output_options['name']
+                expr_lhs = expr.split('=')[0].strip()
+                if '.' in expr_lhs or ':' in expr_lhs:
+                    raise ValueError(
+                        f'{expr_lhs} is an invalid name for the timeseries LHS. Must use a valid variable name')
+                expr_reduced = expr
+
+                units = output_options['units'] if output_options['units'] is not _unspecified else None
+                shape = output_options['shape'] if output_options['shape'] not in {_unspecified, None} else (1,)
+
+                abs_names = [x.strip() for x in re.findall(re.compile(r'([_a-zA-Z]\w*[ ]*\(?:?[.]?)'), expr)
```

Comment:
You might want to do the regex compilation once before the beginning of the outer loop instead of here.